core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -1,27 +1,225 @@
 	 */
-	FrankElement(final String fullName, final String simpleName) {
+	public FrankElement(final String fullName, final String simpleName, boolean isAbstract) {
 		this.fullName = fullName;
 		this.simpleName = simpleName;
+		this.isAbstract = isAbstract;
+		this.aliasSources = new ArrayList<>();
+	}
+
+	public FrankElement(final String fullName, final String simpleName) {
+		this(fullName, simpleName, false);
+	}
+	
+	public void setParent(FrankElement parent) {
+		this.parent = parent;
+		this.statistics = new FrankElementStatistics(this);
+	}
+
+	/**
+	 * Setter for attributes. We prevent modifying the list of attributes
+	 * because we want to maintain the private field attributeLookup.
+	 * @param inputAttributes
+	 */
+	public void setAttributes(List<FrankAttribute> inputAttributes) {
+		Collections.sort(inputAttributes);
+		this.attributes = Collections.unmodifiableList(inputAttributes);
+		attributeLookup = new HashMap<>();
+		for(FrankAttribute a: attributes) {
+			if(attributeLookup.containsKey(a.getName())) {
+				log.warn(String.format("Frank element [%s] has multiple attributes with name [%s]",
+						fullName, a.getName()));
+			} else {
+				attributeLookup.put(a.getName(), a);
+			}
+		}
+	}
+
+	public List<FrankAttribute> getAttributes(Predicate<? super FrankAttribute> filter) {
+		return attributes.stream().filter(filter).collect(Collectors.toList());
+	}
+
+	/**
+	 * Setter for config children. We prevent modifying the list of config children
+	 * because we want to maintain the private field configChildLookup.
+	 * @param children
+	 */
+	public void setConfigChildren(List<ConfigChild> children) {
+		Collections.sort(children);
+		this.configChildren = Collections.unmodifiableList(children);
+		configChildLookup = new HashMap<>();
+		for(ConfigChild c: children) {
+			ConfigChildKey key = new ConfigChildKey(c);
+			if(configChildLookup.containsKey(key)) {
+				log.warn(String.format("Different config children of Frank element [%s] have the same key", fullName));
+			} else {
+				configChildLookup.put(key, c);
+			}
+		}
+	}
+
+	public List<ConfigChild> getConfigChildren(Predicate<? super ConfigChild> filter) {
+		return configChildren.stream().filter(filter).collect(Collectors.toList());
+	}
+
+	ElementChild findElementChildMatch(ElementChild elementChild) {
+		if(elementChild instanceof FrankAttribute) {
+			return findAttributeMatch((FrankAttribute) elementChild);
+		} else if(elementChild instanceof ConfigChild) {
+			return findConfigChildMatch((ConfigChild) elementChild);
+		} else {
+			throw new IllegalArgumentException(String.format(
+					"Expected a FrankAttribute or ConfigChild, but got a [%s]",
+					elementChild.getClass().getName()));
+		}
+	}
+
+	FrankAttribute findAttributeMatch(FrankAttribute attribute) {
+		return attributeLookup.get(attribute.getName());
+	}
+
+	ConfigChild findConfigChildMatch(ConfigChild configChild) {
+		return configChildLookup.get(new ConfigChildKey(configChild));
```

Comment:
Kan ConfigChild niet gewoon een getKey() method hebben?

Reply:
Gedaan. De prijs die we betalen is dat de key velden van ConfigChild in een String gecombineerd moeten worden. Dat was nodig om te zorgen dat ElementChild.getKey() van het type String kan zijn. Ik wilde het type argument K voor de key niet opnieuw invoeren, want ik voorzag veel problemen met type conversies tussen FrankAttribute, ConfigChild, ElementChild<T> en ElementChild<?>

Reply:
Door de telefoon zei je dat je dat ik hier nog mee verder moet. Dat ga ik nog doen.

Reply:
Ik heb mijn best gedaan. Er zijn in FrankElement vrij veel kasts tussen FrankAttribute, ElementChild en ConfigChild. Kan je kijken of je zo tevreden bent?